_rules/autocomplete-valid-value-73f2c2.md

```diff
@@ -169,7 +171,7 @@ This `autocomplete` [attribute value][] is comma separated instead of space usin
 This `autocomplete` [attribute value][] is not appropriate for the field. It is not possible to type an e-mail using just numbers.
 
 ```html
-<label>Email<input type="number" autocomplete="email"/></label>
+<label>Quantity<input type="number" autocomplete="email"/></label>
```

Comment:
Wouldn't that fail the second Assumption? I don't think we should have examples for which the Assumptions don't hold (at least, we don't have any currently as far as I can tell).

Reply:
The second assumption is that incorrect `type` attribute is failing this rule (because of the assumption itself) but it doesn't necessarily fail 1.3.5. which I think makes the failed example correct. I think that both the incorrect `type` attribute or `autocomplete="off"` on the parent `<form>` can prevent browsers from autocompleting the value but they do not prevent the purpose of the widget from being programmatically determinable.

My issue is that this rule has 1.3.5 as the accessibility requirement. On the rule design page, in the [accessibility requirements section,](https://act-rules.github.io/pages/design/rule-design/#accessibility-requirements) it says: 

> List all accessibility requirements that are not satisfied if one or more outcomes of a rule fail. For example, this could be one or more WCAG success criteria, or a WAI-ARIA rule

So, the failed example 5 does not really map to 1.3.5. At least in my opinion.

Reply:
This isn't an example of the new assumption. It's the opposite. That's why the label changed. The label and type suggest it needs a number, but the autocomplete attribute is inconsistent with that. That's a failure of expectation 2. Clearly this is creating some confusion though, so if you have a suggestion on how to improve this, that would be great.

Reply:
🤔 
Let me try and get the detailed explanation…
- For `input` elements, the correctness of the `autocomplete` attribute depends on the type of the element as you can't autocomplete a number the same way as you autocomplete an address.
- A mismatch between `type` and `autocomplete` thus means that one of them is wrong, but there is no way to decide which as it depends on the intention of the author, something we do not have access to.
- Thus, the rule assumes that `type` is correct and can focus on flagging `autocomplete` errors. Hence the new assumption.
- This example had a wrong `type="number"` (with an intention of "email") and is now updated to try and have an intention of "number" (so that `type` is correct and the rule can work).
✔️ I agree this is a correct example.

I think we need to also update the description to make the intention (of this `input`) clear. Especially, "it is not possible to type an e-mail using just numbers" is what lead me to believe that the intention of this `input` was "email" rather than "number".

> This `autocomplete` [attribute value][] is not appropriate for the field. The form field is intended to input a quantity (a number) which cannot be a e-mail.

or something like that, i.e. the problem is not that we can't type an e-mail with numbers but rather than an e-mail will not be composed solely of numbers, same-same-ish, but different in the direction we read the (non)-equivalence. The current description looks like "duh, you're asking me to type my e-mail with numbers!?" and should be moved to "duh, you're taking my e-mail and consider that it is a number!?"

Reply:
Yep, I was focusing too much on the combination between the `type` attribute and the `autocomplete`. I didn't think about the purpose of the `<label>`.

The combination of the correct `<label>`, correct `autocomplete` and incorrect `type` attribute could be mentioned in the expectation.

Reply:
Using Emma's suggestion for the description.